Lib/xmlrpc/server.py

```diff
@@ -387,29 +387,40 @@ def _dispatch(self, method, params):
         """
 
         func = None
+
         try:
             # check to see if a matching function has been registered
             func = self.funcs[method]
         except KeyError:
-            if self.instance is not None:
-                # check for a _dispatch method
-                if hasattr(self.instance, '_dispatch'):
-                    return self.instance._dispatch(method, params)
-                else:
-                    # call instance method directly
-                    try:
-                        func = resolve_dotted_attribute(
-                            self.instance,
-                            method,
-                            self.allow_dotted_names
-                            )
-                    except AttributeError:
-                        pass
+            pass
 
         if func is not None:
             return func(*params)
-        else:
-            raise Exception('method "%s" is not supported' % method)
+
+        if self.instance is not None:
+            try:
+                # check for a _dispatch method on the instance
+                func = self.instance._dispatch
```

Comment:
It would be better to use other name for local variable.

But I suggest to keep the current code, with `hasattr()`. Catching exceptions is costly in Python, and if `_dispatch` is not defined this causes regression.